src/sess.rs

```diff
@@ -178,6 +178,13 @@ impl<'sess, 'ctx: 'sess> Session<'ctx> {
             let mut ranks: HashMap<DependencyRef, usize> =
                 graph.keys().map(|&id| (id, 0)).collect();
             let mut pending = HashSet::new();
+            for name in self.manifest.dependencies.keys() {
+                if !(names.contains_key(name)) {
+                    return Err(Error::new(format!(
+                        "There is a mismatch in the lock file, you may need to run `bender update`."
```

Comment:
I would state this more explicitly; for example:
```suggestion
                        "`Bender.yml` contains dependency `{}` but `Bender.lock` does not.  You may need to run `bender update`.", name
```